coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+          ``output_regex``.
+        - ``'corrected'``: The output is the corrected of the given file. Diffs
+          are then generated to supply patches for results.
+
+        Passing something else raises a ``ValueError``.
+    :param output_regex:
+        The regex expression as a string that is used to parse the output
+        generated by the underlying executable. It should use as many of the
+        following named groups (via ``(?P<name>...)``) to provide a good
+        result:
+
+        - line - The line where the issue starts.
+        - column - The column where the issue starts.
+        - end_line - The line where the issue ends.
+        - end_column - The column where the issue ends.
+        - severity - The severity of the issue.
+        - message - The message of the result.
+        - origin - The origin of the issue.
+
+        The groups ``line``, ``column``, ``end_line`` and ``end_column`` don't
+        have to match numbers only, they can also match nothing, the generated
+        ``Result`` is filled automatically with ``None`` then for the
+        appropriate properties.
+
+        Needs to be provided if ``output_format`` is ``'regex'``.
+    :param severity_map:
+        A dict used to map a severity string (captured from the
+        ``output_regex`` with the named group ``severity``) to an actual
+        ``coalib.results.RESULT_SEVERITY`` for a result. Severity strings are
+        mapped **case-insensitive**!
+
+        - ``RESULT_SEVERITY.MAJOR``: Mapped by ``error``.
+        - ``RESULT_SEVERITY.NORMAL``: Mapped by ``warning`` or ``warn``.
+        - ``RESULT_SEVERITY.MINOR``: Mapped by ``info``.
+
+        A ``ValueError`` is raised when the named group ``severity`` is not
+        used inside ``output_regex`` and this parameter is given.
+    :param diff_severity:
+        The severity to use for all results if ``output_format`` is
+        ``'corrected'``. By default this value is
+        ``coalib.results.RESULT_SEVERITY.NORMAL``. The given value needs to be
+        defined inside ``coalib.results.RESULT_SEVERITY``.
+    :param diff_message:
+        The message-string to use for all results if ``output_format`` is
+        ``'corrected'``. By default this value is ``"Inconsistency found."``.
+    :raises ValueError:
+        Raised when invalid options are supplied.
+    :raises TypeError:
+        Raised when incompatible types are supplied.
+        See parameter documentations for allowed types.
+    :return:
+        A ``LocalBear`` derivation that lints code using an external tool.
+    """
+    options["executable"] = executable
+    options["output_format"] = output_format
+    options["use_stdin"] = use_stdin
+    options["use_stdout"] = use_stdout
+    options["use_stderr"] = use_stderr
+    options["config_suffix"] = config_suffix
+    options["prerequisite_check_command"] = prerequisite_check_command
+
+    _prepare_options(options)
+
+    def create_linter(klass):
```

Comment:
Did you see what I wrote above ? We can make a arg to create_linter and do functools.partial inside Linter to give that arg